lib/storage/providers/LocalForage.js

```diff
@@ -75,10 +75,17 @@ const provider = {
     },
 
     /**
-     * Clear absolutely everything from storage
+     * Clear most things from storage
+     * @param {Array} keyValuesToPreserve a set of key value pairs to keep in Onyx after doing a full clear. This argument
+     *      is here to provide a performance boost when deleting a majority of keys, but wanting to keep only a few.
+     *      Example: It's faster than doing a big multiRemove() which is just a loop that calls removeItem().
      * @returns {Promise<void>}
      */
-    clear: localforage.clear,
+    clear(keyValuesToPreserve) {
+        this.setItemQueue.abort();
+        return localforage.clear()
+            .then(() => Storage.multiSet(keyValuesToPreserve));
+    },
```

Comment:
Following my prior comment this can become 

```js
    clear(keysToPreserve) {
        this.setItemQueue.abort();
        return this.multiGet(keysToPreserve)
            .then(keyValuesToPreserve =>
                localforage.clear()
                    .then(() => this.multiSet(keyValuesToPreserve))
             )
    },
```